plotly/offline.py

```diff
@@ -0,0 +1,178 @@
+    Run this function at the start of an IPython notebook
+    to load the necessary javascript files for creating
+    Plotly graphs with plotly.offline.iplot.
+    """
+    from IPython.display import HTML, display
+
+    if not os.path.exists(PLOTLY_OFFLINE_BUNDLE):
+        raise PlotlyError('Plotly Offline source file at {source_path} '
+                          'is not found.\n'
+                          'If you have a Plotly Offline license, then try '
+                          'running plotly.offline.download_plotlyjs(url) '
+                          'with a licensed download url.\n'
+                          "Don't have a Plotly Offline license? "
+                          'Contact sales@plot.ly learn more about licensing.\n'
+                          'Questions? support@plot.ly.'
+                          .format(source_path=PLOTLY_OFFLINE_BUNDLE))
+
+    global __PLOTLY_OFFLINE_INITIALIZED
+    __PLOTLY_OFFLINE_INITIALIZED = True
+    display(HTML('<script type="text/javascript">' +
+                 open(PLOTLY_OFFLINE_BUNDLE).read() + '</script>'))
+
+
+def iplot(figure_or_data, show_link=True, link_text='Export to plot.ly'):
+    """
+    Draw plotly graphs inside an IPython notebook without
+    connecting to an external server.
+    To save the chart to Plotly Cloud or Plotly Enterprise, use
+    `plotly.plotly.iplot`.
+    To embed an image of the chart, use `plotly.image.ishow`.
+
+    figure_or_data -- a plotly.graph_objs.Figure or plotly.graph_objs.Data or
+                      dict or list that describes a Plotly graph.
+                      See https://plot.ly/python/ for examples of
+                      graph descriptions.
+
+    Keyword arguments:
+    show_link (default=True) -- display a link in the bottom-right corner of
+                                of the chart that will export the chart to
+                                Plotly Cloud or Plotly Enterprise
+    link_text (default='Export to plot.ly') -- the text of export link
+
+    Example:
+    ```
+    from plotly.offline import init_notebook_mode, iplot
+    init_notebook_mode()
+
+    iplot([{'x': [1, 2, 3], 'y': [5, 2, 7]}])
+    ```
+    """
+    if not __PLOTLY_OFFLINE_INITIALIZED:
+        raise PlotlyError('\n'.join([
+            'Plotly Offline mode has not been initialized in this notebook. '
+            'Run: ',
+            '',
+            'import plotly',
+            'plotly.offline.init_notebook_mode() '
+            '# run at the start of every ipython notebook',
+        ]))
+    from IPython.display import HTML, display
+    if isinstance(figure_or_data, dict):
+        data = figure_or_data['data']
+        layout = figure_or_data.get('layout', {})
+    else:
+        data = figure_or_data
+        layout = {}
+
+    width = layout.get('width', '100%')
+    height = layout.get('height', 525)
+    try:
+        float(width)
+        width = str(width) + 'px'
+    except:
```

Comment:
:cow2: i think you don't need such a broad exception clause, can we just look before we leap on this one? or, catch `(ValueError, TypeError)`? OR, use this format perhaps?

```
try:
   float(width)
except:
    pass
else:
    width = str(width) + 'px'
```